src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -1249,6 +1249,22 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
             break;
         }
 
+        case NI_Vector512_ExtractMostSignificantBits:
+        {
+            if (compOpportunisticallyDependsOn(InstructionSet_AVX512F) &&
+                compOpportunisticallyDependsOn(InstructionSet_AVX512BW) &&
+                compOpportunisticallyDependsOn(InstructionSet_AVX512DQ))
```

Comment:
We've been using `IsBaselineVector512IsaSupported` to handle this, makes it a bit simpler and allows us to optimize it where possible.